pkg/virt-controller/services/renderresources.go

```diff
@@ -142,6 +142,35 @@ func WithMemoryOverhead(guestResourceSpec v1.ResourceRequirements, memoryOverhea
 	}
 }
 
+func WithCPUPinning(cpu *v1.CPU) ResourceRendererOption {
+	return func(renderer *ResourceRenderer) {
+		vcpus := hardware.GetNumberOfVCPUs(cpu)
+		if vcpus != 0 {
+			renderer.calculatedLimits[k8sv1.ResourceCPU] = *resource.NewQuantity(vcpus, resource.BinarySI)
+		} else {
+			if cpuLimit, ok := renderer.vmLimits[k8sv1.ResourceCPU]; ok {
+				renderer.vmRequests[k8sv1.ResourceCPU] = cpuLimit
+			} else if cpuRequest, ok := renderer.vmRequests[k8sv1.ResourceCPU]; ok {
+				renderer.vmLimits[k8sv1.ResourceCPU] = cpuRequest
+			}
```

Comment:
I am just copying code from what it had before - or am trying to. 

Do you think I am deviating from the original behavior ? I think originally this would also overwrite the user provided requests.

Reply:
Well, it can be fixed in a future PR